lms/djangoapps/courseware/docs/decisions/0004-personalized-relative-dates.rst

```diff
@@ -0,0 +1,89 @@
+0004. Personalized/Relative Dates in Courseware
+***********************************************
+
+~~~~~~~
+Context
+~~~~~~~
+
+Self-paced courses are, based on the experience of users on edx.org,
+less engaging and less likely to be completed, than instructor-paced courses.
+Research shows that learners are better able to complete a course of study
+when that course gives them multiple incremental deadlines along the way,
+rather than one single deadline at the end of the course. Unfortunately,
+self-paced courses are currently only able to offer the latter experience.
+
+``edx-when`` is a library currently used by ``edx-platform`` to provide a
+read-optimized view of the dates associated with course content. It is updated
+when a course is published, and stores dates in a dedicated table in the
+relational database.
+
+~~~~~~~~
+Decision
+~~~~~~~~
+
+In order to support incremental deadlines in self-paced courses, we want
+to extend edx-platform with the following capabilities:
+
+1.  Extend the api provided by ``edx-when`` used in the LMS to so that
+    the LMS can ask for dates for students (supplying their ``Schedule``), and
+    ``edx-when`` will return an absolute date (even if the date defined by
+    Studio is relative).
+2.  When a self-paced course is published, add relative due dates to it,
+    evenly spaced across the expected duration of the course, unless the
+    course already has relative dates attached to content.
```

Comment:
`already has relative dates attached` - drop the word `relative` maybe?

Reply:
Actually, I think studio doesn't strip the absolute dates from a instructor-paced course if you convert it to self-paced, so we only want to look for relative dates.